### PR TITLE
Drop iot-gc in favor of finalizers

### PR DESCRIPTION
**Pull request by ctron**

### CODE REVIEW

**Comment by ctron** [comment]
@enmasse-ci run tests profile=shared-iot

**Comment by ctron** [comment]
@enmasse-ci run tests profile=shared-iot

File: pkg/controller/iotconfig/collector.go [hunk 1]
```diff
@@ -8,46 +8,24 @@ package iotconfig
...
-		}
-
-		// set default resource limits
 
-		container.Resources = corev1.ResourceRequirements{
-			Limits: corev1.ResourceList{
-				corev1.ResourceMemory: *resource.NewQuantity(128*1024*1024 /* 128Mi */, resource.BinarySI),
-			},
-		}
+	// we no longer need the "iot-gc", so we only remove it
```
Comment (dejanb) [inline pkg/controller/iotconfig/collector.go]: Why do we need this? To deal with upgrades?

Comment (ctron) [reply pkg/controller/iotconfig/collector.go]: Yes. When yo perform an upgrade, it will remove the deployment.

**Pull request merged**